### PR TITLE
Change `attrs.names` to `attrs.dnodes`

### PR DESCRIPTION
**Pull request by jqmp**
This is the first step to allowing a provider to output a tuple
descriptor rather than an entity descriptor. 

Later, once we fully support tuple descriptors, each provider will
have just one output descriptor, so it will be `attrs.dnode`, not
`attrs.dnodes`.

### CODE REVIEW

File: bionic/flow.py [hunk 1]
```diff
@@ -577,6 +579,7 @@ def full_name(first_name, last_name):
         state = self._state
 
         names = [name for name, value in name_value_pairs]
+        print(names)
```
Comment (namanjain) [inline bionic/flow.py:582]: remove print statement. Maybe we should add a flake8 check for print statements too.

File: bionic/provider.py [hunk 2]
```diff
@@ -90,27 +90,38 @@ def get_source_func(self):
...
         return ValueProvider(names)
 
     def __init__(self, names):
+        dnodes = [entity_dnode_from_descriptor(name) for name in names]
         super(ValueProvider, self).__init__(
-            attrs=ProviderAttributes(names=names, changes_per_run=False,),
+            attrs=ProviderAttributes(dnodes=dnodes, changes_per_run=False,),
         )
 
+        self.names = names
```
Comment (namanjain) [inline bionic/provider.py:124]: Wdyt about using ``self.attrs.entity_names` instead of storing the names here?

File: bionic/flow.py (hunk 1 above)
Comment (jqmp) [reply bionic/flow.py:582]: Great idea. It looks like [flake8-print](https://pypi.org/project/flake8-print/) does this -- I'll see if I can set it up. (I also want to check out [flake8-fixme](https://pypi.org/project/flake8-fixme/), which addresses another common annoyance of mine.)

File: bionic/provider.py (hunk 2 above)
Comment (jqmp) [reply bionic/provider.py:124]: I don't feel strongly either way -- happy to change it.

**Pull request merged**